File: lib/src/orders/domain/entities/order.dart
import 'package:equatable/equatable.dart';

class Order extends Equatable {
  final String number;
  final String type;
  final String arrivalDate;
  final String secretary;
  final String project;
  final String description;
  final String sendDate;
  final String returnDate;
  final String situation;
  final String notes;
  final bool isArchived;

  const Order({
    required this.number,
    required this.type,
    required this.arrivalDate,
    required this.secretary,
    required this.project,
    required this.description,
    this.sendDate = '',
    this.returnDate = '',
    this.situation = '',
    this.notes = '',
    this.isArchived = false,
  });

  @override
  List<Object?> get props => [
        number,
        type,
        arrivalDate,
        secretary,
        project,
        description,
        sendDate,
        returnDate,
        situation,
        notes,
        isArchived,
      ];
}

File: src/features/orders/types/order.ts
import { OrderType } from 'src/features/orders/types/order-type';

export interface Order {
  number: string;
  type: OrderType;
  arrivalDate: string;
  secretary: string;
  project: string;
  description: string;
  sendDate: string;
  returnDate: string;
  situation: string;
  notes: string;
  isArchived?: boolean;
}

export const emptyOrder = (): Order => ({
  number: '',
  type: OrderType.SE,
  arrivalDate: '',
  secretary: '',
  project: '',
  description: '',
  sendDate: '',
  returnDate: '',
  situation: '',
  notes: '',
});
